XenAdmin/SettingsPanels/PerfmonAlertOptionsPage.cs

```diff
@@ -107,33 +113,74 @@ private bool IsValidSmtpAddress()
             return !SmtpServerAddrTextBox.Text.ToCharArray().Any((c) => c >= 128) && SmtpServerAddrTextBox.Text.Trim().Length > 0;
         }
 
+        public void SetMailLanguageComboBoxValue(String code)
+        {
+            if (_bSupportMailLanguage)
+            {   // Feature supported
+                if (null == code)
+                {   // Set default value
+                    if (PerfmonOptionsDefinition.MailLanguageHasCode(Branding.BRANDING_PERF_ALERT_MAIL_LANGUAGE_DEFAULT))
+                        MailLanguageComboBox.SelectedValue = Branding.BRANDING_PERF_ALERT_MAIL_LANGUAGE_DEFAULT;
+                    else
+                        MailLanguageComboBox.SelectedIndex = 0;
+                }
+                else if (PerfmonOptionsDefinition.MailLanguageHasCode(code))
+                    MailLanguageComboBox.SelectedValue = code;  // Valid language code, set it to MailLanguageComboBox
+                else
+                    MailLanguageComboBox.SelectedValue = null;  // Invalid language code, set null to MailLanguageComboBox
+            }
+            else
+                MailLanguageComboBox.SelectedValue = null;  // Feature not supported
+        }
+
         public void SetXenObjects(IXenObject orig, IXenObject clone)
         {
             _XenModelObject = clone;
 
+            if (Helpers.FalconOrGreater(_XenModelObject.Connection))
+                _bSupportMailLanguage = true;
+            else
+                _bSupportMailLanguage = false;
+
             Repopulate();
 
             // Save original settings for change detection
             _OrigEmailNotificationCheckBox = EmailNotificationCheckBox.Checked;
             _OrigEmailAddressTextBox = EmailAddressTextBox.Text;
             _OrigSmtpServerAddrTextBox = SmtpServerAddrTextBox.Text;
             _OrigSmtpServerPortTextBox = SmtpServerPortTextBox.Text;
+
+            if (_bSupportMailLanguage)
+                _OrigMailLanguageCodeComboBox = null == MailLanguageComboBox.SelectedValue ? null : MailLanguageComboBox.SelectedValue.ToString();

```

Comment:
I think that the _OrigMailLanguageCodeComboBox should contain the _PerfmonOptions.MailLanguageCode as it is saved on the pool, not combobox's selected value, which in many cases is defaulted to something. because we nned to save it even in the case when the user has not changed the default value.